### PR TITLE
optionally enable export if not exported model provided

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@echarlaix @AlexKoff88 @helena-intel could you please review and merge before next optimum release?

File: optimum/intel/openvino/modeling_base.py [hunk 1]
```diff
@@ -524,3 +526,156 @@ def can_generate(self) -> bool:
...
+                config,
+                revision=revision,
+                cache_dir=cache_dir,
+                use_auth_token=use_auth_token,
+                force_download=force_download,
+                subfolder=subfolder,
+                trust_remote_code=trust_remote_code,
+            )
+
+        if export is None:
```
Comment (AlexKoff88) [inline optimum/intel/openvino/modeling_base.py:610]: A better way to introduce a separate variable  and do the same like:
```
need_export = export or cls._check_export_status(model_id, revision, subfolder, cache_dir, local_files_only or HF_HUB_OFFLINE)
```

**Comment by AlexKoff88** [comment]
I don't have strong objections but I wonder if there is a more elegant way to do the same without introducing so much code that mostly duplicated Transformers logic.  @echarlaix should know it better.

Also a test is required.

**Comment by echarlaix** [comment]
> I don't have strong objections but I wonder if there is a more elegant way to do the same without introducing so much code that mostly duplicated Transformers logic. @echarlaix should know it better.
> 
> Also a test is required.

Yes I think we can simplify this by using [find_files_matching_pattern](https://github.com/huggingface/optimum/blob/7a0757aeeb7a5a09d0f4e68aadfe0654277a0298/optimum/utils/file_utils.py#L41)  like :
```python
pattern = r"(.*)?openvino(.*)?\_model.xml"
ov_files = find_files_matching_pattern(
    model,
    pattern,
    subfolder=subfolder,
    use_auth_token=token,
    revision=revision,
)

export = len(ov_files) == 0
```

also used in https://github.com/huggingface/optimum-intel/pull/740 for the openvino pipelines

**Pull request closed without merging**